### PR TITLE
Address pilot namespace comment when running in non istio-system ns #7470

### PR DESCRIPTION
**Pull request by linsun**
@costinm @rshriram upon further testing, I found an issue with the gateway in a different namespace as it keeps restarting due to the  envoyStatsd and zipkin being in a different namespace.

would be great to get this in 1.0 if we are spinning a new build today.

### CODE REVIEW

**Comment by linsun** [comment]
With this change, no restart on the pods in kube-system now.

```
$ kubectl get pods --all-namespaces | grep gateway      
istio-system   istio-egressgateway-868dcd79f-k6jwf                               1/1       Running   0          1h
istio-system   istio-ingressgateway-6d75ffcf78-nxngn                             1/1       Running   0          1h
kube-system    istio-egressgateway-6879fffdf9-t8psz                              1/1       Running   0          54m
kube-system    istio-ingressgateway-5565dc5fd6-fdw9x                             1/1       Running   0          54m

```

**Comment by linsun** [comment]
@sebastien-prudhomme sorry I didn't see yours earlier.  Is yours tested with 2nd gateway in a different ns?

**Comment by linsun** [comment]
BTW, upon further testing and working with @frankbu, I validated that the 2nd gateway did work, however, user needs to specify in the gateway resource to the matching selector that matches the 2nd gateway.

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -113,6 +113,9 @@ global:
   # If not set, controller watches all namespaces
   oneNamespace: false
 
+  # istio control plane namespace
+  istioNamespace: istio-system
+
```
Comment (rshriram) [inline install/kubernetes/helm/istio/values.yaml]: ? This doesn't seem right.. What is the .Release.namespace for ?
Also, why have this variable here when its not used by anything?

Put another way, if we install istio in a different namespace, how will someone know to update this variable as well?

File: install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml [hunk 2]
```diff
@@ -54,7 +54,7 @@ spec:
           - --serviceCluster
           - {{ $key }}
           - --zipkinAddress
-          - zipkin:9411
+          - zipkin.{{ $.Values.global.istioNamespace }}:9411
```
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: this should have a different condition just like what @objectiser has in his PR

File: install/kubernetes/helm/istio/values.yaml (hunk 1 above)
Comment (linsun) [reply install/kubernetes/helm/istio/values.yaml]: it is only used by the gateways to indicate so the gateway knows where the control plane runs.  (Today's current code hardcoded it to istio-system, with this it is configurable.). Yes, user does need to update this as well for very advanced installation where control plane is installed in a namespace other than istio-system.

Comment (costinm) [reply install/kubernetes/helm/istio/values.yaml]: Release.namespace is where gateway is installed.
istioNamespace is where istio control plane is installed.
They may be different.

This is reusing the chart for ingress for installing ingress in different namespace.

File: install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml [hunk 3]
```diff
@@ -54,9 +54,13 @@ spec:
...
           - {{ $key }}
           - --zipkinAddress
+        {{- if $.Values.global.istioNamespace }}
+          - zipkin.{{ $.Values.global.istioNamespace }}:9411
+        {{- else }}
           - zipkin:9411
-        {{- if $.Values.global.proxy.envoyStatsd.enabled }}
+        {{ end }}
           - --statsdUdpAddress
+        {{- if $.Values.global.proxy.envoyStatsd.enabled }}
```
Comment (costinm) [inline install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: that's bad - need to capture the line above

File: install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml [hunk 4]
```diff
@@ -54,9 +54,13 @@ spec:
           - --serviceCluster
           - {{ $key }}
           - --zipkinAddress
+        {{- if $.Values.global.istioNamespace }}
+          - zipkin.{{ $.Values.global.istioNamespace }}:9411
+        {{- else }}
           - zipkin:9411
-        {{- if $.Values.global.proxy.envoyStatsd.enabled }}
+        {{- end }}
           - --statsdUdpAddress
```
Comment (rshriram) [inline install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: this is wrong.. you need the condition above statsdUdpAddress

File: install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml [hunk 5]
```diff
@@ -65,12 +69,20 @@ spec:
           - --controlPlaneAuthPolicy
           - MUTUAL_TLS
           - --discoveryAddress
-          - istio-pilot.istio-system:15005
+          {{- if $.Values.global.istioNamespace }}
+          - istio-pilot.{{ $.Values.global.istioNamespace }}:15005
```
Comment (gyliu513) [inline install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: @linsun I was a bit confused here, why using 15005 for both mTLS and non-mTLS mode?

Comment (gyliu513) [reply install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: I created a PR at https://github.com/istio/istio/pull/7520 to fix this.

**Pull request merged**